### DISCUSSION TITLE
DB dump should indicate which version readme, etc. came from

### DISCUSSION THREAD

**Discussion by jmyersmsft** [rust-lang/crates.io#5346] [Feature Requests]
I've been working on some code that compares the data available on crates.io with the contents of the actual .crate files. I've been using the db dump (along with the index), but for readme, etc. it only contains the data for a single version. This is fine for now as I'm currently only interested in identifying categories of discrepancies, but there's no indication of which version that information came from, complicating matching. For most crates, I can match the crate's updated_at with a version's created_at, but that doesn't work for all crates. It's not immediately clear how the version to provide that information is selected (highest version? most recently published?). I'd prefer to avoid parsing the version strings, if possible, but I can if necessary.

It would be nice if the crates table in the dump had an indication of which version the readme, etc. came from. This could be the version string or the foreign key to the versions table, doesn't matter to me.

**Comment by Turbo87** [comment]
in the current implementation the crate README in the database is overwritten by each new `cargo publish`, independent from the version number. there are thoughts about changing that in the future, but for now it should be sufficient to sort the versions by `created_at` and pick the newest one.

**Reply by jmyersmsft** [reply]
It'd still be nice to have the indication of the version for completeness, but that detail should get me what I need for now. Thanks!

**Reply by Turbo87** [reply]
yeah, as I said, there are ideas on how we can make this more straight forward, they just don't have the highest priority right now.
